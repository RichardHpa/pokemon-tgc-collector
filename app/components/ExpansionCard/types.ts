import type { Set } from "pokemon-tcg-sdk-typescript/dist/sdk";

export interface ExpansionCardProps
  extends Pick<Set, "name" | "series" | "releaseDate"> {
  image: string;
}
