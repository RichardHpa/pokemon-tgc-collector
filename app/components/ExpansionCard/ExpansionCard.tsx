import { Card } from "~/components/Card";

import type { ExpansionCardProps } from "./types";

export const ExpansionCard = ({
  image,
  name,
  series,
  releaseDate,
}: ExpansionCardProps) => {
  return (
    <Card>
      <div className="flex h-full flex-col">
        <div className="mb-4 flex justify-center">
          <img className="max-h-28 w-auto" src={image} alt={`${name} logo`} />
        </div>

        <div className="mt-auto">
          <h5 className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white">
            {name}
          </h5>

          <p className="mb-3 font-normal text-gray-700 dark:text-gray-400">
            {series} - {releaseDate}
          </p>
        </div>
      </div>
    </Card>
  );
};
