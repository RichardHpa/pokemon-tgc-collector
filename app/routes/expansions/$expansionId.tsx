import type { ActionArgs, LoaderArgs } from "@remix-run/node";
import { json, redirect } from "@remix-run/node";
import { useLoaderData } from "@remix-run/react";
import invariant from "tiny-invariant";

import { getExpansion } from "~/api/getExpansion";

type LoaderData = {
  expansion: Awaited<ReturnType<typeof getExpansion>>;
};

export async function loader({ params }: LoaderArgs) {
  invariant(params.expansionId, "expansionId not found");

  const data = await getExpansion(params.expansionId);

  return json<LoaderData>({
    expansion: data,
  });
}

export default function ExpansionDetailsPage() {
  const { expansion } = useLoaderData<typeof loader>();
  console.log(expansion);
  return (
    <div className="grid grid-flow-row-dense grid-cols-3 gap-4">
      <img src={expansion.images.logo} alt={`${expansion.name} logo`} />

      <div className="col-span-2">
        <h3 className="mb-2 text-2xl font-bold">
          {expansion.name} {expansion.ptcgoCode && `(${expansion.ptcgoCode})`}
        </h3>
        {expansion.series && (
          <p className="mb-2 font-bold">{expansion.series}</p>
        )}
        {expansion.releaseDate && (
          <p className="mb-2 font-bold">released {expansion.releaseDate}</p>
        )}

        <img
          className="w-12"
          src={expansion.images.symbol}
          alt={`${expansion.name} symbol`}
        />
      </div>
    </div>
  );
}
