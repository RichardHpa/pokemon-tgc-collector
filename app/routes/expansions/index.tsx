import { useLoaderData } from "@remix-run/react";
import { json } from "@remix-run/node";

import { getExpansions } from "~/api/getExpansions";

import { ExpansionCard } from "~/components/ExpansionCard";

import type { LoaderFunction } from "@remix-run/node";

type LoaderData = {
  data: Awaited<ReturnType<typeof getExpansions>>;
};

export const loader: LoaderFunction = async () => {
  return json<LoaderData>({
    data: await getExpansions({ orderBy: "releaseDate", pageSize: 12 }),
  });
};

export default function ExpansionsIndexPage() {
  const data = useLoaderData<LoaderData>();

  return (
    <div>
      {data?.data && (
        <div className="grid grid-cols-4 gap-3" role="list">
          {data?.data.map((set) => {
            return (
              <div
                key={set.id}
                className="flex-1 items-stretch"
                role="listitem"
              >
                <ExpansionCard
                  name={set.name}
                  series={set.series}
                  releaseDate={set.releaseDate}
                  image={set.images.logo}
                />
              </div>
            );
          })}
        </div>
      )}
    </div>
  );
}
