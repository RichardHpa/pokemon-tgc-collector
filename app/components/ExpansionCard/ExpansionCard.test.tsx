import { render, screen } from "@testing-library/react";

import { expansion } from "mocks/fixtures/expansions";

import { ExpansionCard } from "./ExpansionCard";

const props = {
  name: expansion.name,
  releaseDate: expansion.releaseDate,
  series: expansion.series,
  image: expansion.images.logo,
};

describe("ExpansionCard", () => {
  test("renders ExpansionCard", () => {
    render(<ExpansionCard {...props} />);
    expect(screen.getByText(expansion.name)).toBeInTheDocument();
    expect(
      screen.getByText(`${expansion.series} - ${expansion.releaseDate}`)
    ).toBeInTheDocument();
  });

  test("renders SetCard snapshot", () => {
    const { container } = render(<ExpansionCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
