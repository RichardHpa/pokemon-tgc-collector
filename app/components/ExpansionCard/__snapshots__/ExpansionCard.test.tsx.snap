// Vitest Snapshot v1

exports[`ExpansionCard > renders SetCard snapshot 1`] = `
<div>
  <div
    class="h-full rounded-lg border border-gray-200 bg-white p-4 shadow-md dark:border-gray-700 dark:bg-gray-800"
  >
    <div
      class="flex h-full flex-col"
    >
      <div
        class="mb-4 flex justify-center"
      >
        <img
          alt="Lost Origin logo"
          class="w-full max-w-xs"
          src="https://images.pokemontcg.io/swsh11/logo.png"
        />
      </div>
      <div
        class="mt-auto"
      >
        <h5
          class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
        >
          Lost Origin
        </h5>
        <p
          class="mb-3 font-normal text-gray-700 dark:text-gray-400"
        >
          Sword & Shield
           - 
          2022/09/09
        </p>
      </div>
    </div>
  </div>
</div>
`;
